### PR TITLE
Added Bling Subsystem #27

### LINKED ISSUE #27: Bling Subsystem
Controlling colours for:
- fun
- driver feedback

Involves writing to network tables from roboRio to communicate with raspberry Pi

Definition of Done:
Production software for the bling subsystem in the robot code, including:
- interface definitions
- commands
- etc..
- tested independently
- merged and integrated with 2020 robot code in github
- ready for integration

### PR DESCRIPTION
**Pull request by jwt2706**
It's just the frame work.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Bling.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Bling extends SubsystemBase {
+
+
+  private static Bling INSTANCE = new Bling();
```
Comment (nicojopr) [inline src/main/java/frc/robot/subsystems/Bling.java:15]: Suggest you follow the Singleton pattern and move the instantiation ("new Bling()") inside the getInstance or constructor

File: src/main/java/frc/robot/subsystems/Bling.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  /**
+   * Creates a new Bling.
+   */
+  private Bling() {
+
+
+  }
+
+  public static Bling getINSTANCE() {
```
Comment (nicojopr) [inline src/main/java/frc/robot/subsystems/Bling.java:24]: Typically the getINSTANCE will check for null INSTANCE (and create the instance in this method if null) before returning INSTANCE

**Review by kevlam2706 — APPROVED** [review]
🤷🏻‍♂️😂

Not much to review! But maybe good to get the framework into master quickly so you can work on it more later.

Update: oops maybe I read too quickly. Please address Johanna’s comments

**Review by nicojopr — CHANGES_REQUESTED** [review]
At least the null pointer check in getINSTANCE should be addressed before merging

File: src/main/java/frc/robot/subsystems/Bling.java (hunk 1 above)
Comment (jwt2706) [reply src/main/java/frc/robot/subsystems/Bling.java:15]: I originally did that, but when I got Nathan to have a look at it, he said that I could move it there.

File: src/main/java/frc/robot/subsystems/Bling.java (hunk 2 above)
Comment (jwt2706) [reply src/main/java/frc/robot/subsystems/Bling.java:24]: Oh, I see... I haven't done that for shuffleboard either, should I change it on both?

**Review by nicojopr — APPROVED** [review]
Discussed the getINSTANCE risk of non-null INSTANCE, will accept as is for singleton

**Pull request merged**